torch/_C/__init__.pyi.in

```diff
@@ -96,8 +96,6 @@ class device:
     type: str  # THPDevice_type
     index: _int  # THPDevice_index
 
-    def __get__(self, instance, owner=None) -> device: ...
```

Comment:
`tp_descr_get` slot is [set](https://github.com/nenb/pytorch/blob/0de2a45a48b1b97860c4281cc491ee161419e7c9/torch/csrc/Device.cpp#L274) to nullptr, which means that the `device` class does not in fact implement the descriptor protocol, hence why it's removed here.